shell/platform/linux/fl_task_runner.cc

```diff
@@ -0,0 +1,221 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_task_runner.h"
+#include "flutter/shell/platform/linux/fl_engine_private.h"
+
+static constexpr int kMicrosecondsPerNanosecond = 1000;
+static constexpr int kMillisecondsPerMicrosecond = 1000;
+
+struct _FlTaskRunner {
+  GObject parent_instance;
+
+  FlEngine* engine;
+
+  GMutex mutex;
+  GCond cond;
+
+  guint timeout_source_id;
+  GList /*<FlTaskRunnerTask>*/* pending_tasks;
+  gboolean blocking_main_thread;
+
+  gboolean stopped;
+};
+
+typedef struct _FlTaskRunnerTask {
+  // absolute time of task (based on g_get_monotonic_time)
+  gint64 task_time_micros;
+  FlutterTask task;
+} FlTaskRunnerTask;
+
+G_DEFINE_TYPE(FlTaskRunner, fl_task_runner, G_TYPE_OBJECT)
+
+static void fl_task_runner_dispose(GObject* object);
```

Comment:
Can you re-order the functions to be:
- private functions
- class functions (dispose)
- class_init
- init
- public functions

This is for consistency with the other modules and avoids the need for the prototype.

Reply:
Sure